Initial commit for Tag-Repos job

AdamBrousseau opened the pull request:
- Pipeline file to tag OpenJ9 and OpenJ9-OMR
- Fail if Tag already exists
- Fail if tag point does not exist
- Inputs: Tag Name, Tag Annotation (Optional), OpenJ9 SHA, OMR SHA

[skip ci]

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

AdamBrousseau commented:
[Job](https://ci.eclipse.org/openj9/job/Tag-Repos)

AdamBrousseau commented on buildenv/jenkins/jobs/infrastructure/tagRepos:
```diff
@@ -0,0 +1,138 @@
+
+HTTP = 'https://'
+OMR_REPO = 'github.com/AdamBrousseau/openj9-omr.git'
+OPENJ9_REPO = 'github.com/AdamBrousseau/openj9.git'
```
Will update this to `eclipse` once done testing

AdamBrousseau commented on buildenv/jenkins/jobs/infrastructure/tagRepos:
```diff
@@ -0,0 +1,138 @@
+            )
+        echo "SHA: ${SHA}"
+        // This 'if' condition is WIP
+        if ((TAG_POINT ==~ /.*SHA.*/) || (SHA ==~ /.*TAG_POINT.*/)) {
```
This is the only WIP left for me

AdamBrousseau commented:
Test cases
- No Tag Name: https://ci.eclipse.org/openj9/job/Tag-Repos/2
- Default Branches: https://ci.eclipse.org/openj9/job/Tag-Repos/21/
- SHAS & Annotation: https://ci.eclipse.org/openj9/job/Tag-Repos/22/
- 1 SHA, 1 Branch, no Annotation: https://ci.eclipse.org/openj9/job/Tag-Repos/23/

Ready for review @pshipton

AdamBrousseau commented:
@DanHeidinga do we want to make the Tag Annotation mandatory? Do we have a set format for the annotation? The code would be slightly less complicated if it was mandatory. I'm not sure if the annotation makes a difference from a git checkout perspective.

keithc-ca commented:
The command we're using (`git describe --exact-match`) will only work with annotated tags (i.e those created by using `git tag -a ...`).

DanHeidinga commented:
> @DanHeidinga do we want to make the Tag Annotation mandatory?

Yes, the release.md file says to use annotated tags as GITHUB treats them differently than simple tags.  The format should be free form text.

See https://github.com/eclipse/openj9/issues/1265 for examples of the annotation I've used for the 0.8.0 release.

AdamBrousseau commented:
Updated to make ANNOTATION mandatory.
Tests
- no annotaion provided: https://ci.eclipse.org/openj9/job/Tag-Repos/30
- 1 sha, 1 branch: https://ci.eclipse.org/openj9/job/Tag-Repos/29
- only tag name and annotaiton (default branches): https://ci.eclipse.org/openj9/job/Tag-Repos/31

keithc-ca commented on buildenv/jenkins/jobs/infrastructure/tagRepos:
```diff
@@ -0,0 +1,134 @@
+ *******************************************************************************/
+
+// Assumes OPENJ9_SHA, OMR_SHA, and TAG_NAME are passed in Jenkins parameters
+// TAG_ANNOTATION is optional
```
This isn't true anymore.

AdamBrousseau replied:
updated

keithc-ca commented on buildenv/jenkins/jobs/infrastructure/tagRepos:
```diff
@@ -0,0 +1,134 @@
+        def SHA = sh (
+                script: "git rev-parse --short ${POINT_TYPE_PREFIX}${TAG_POINT}",
+                returnStdout: true
+            )
```
Nit: can you adjust the indentation to be consistent with the rest of this file?

keithc-ca commented on buildenv/jenkins/jobs/infrastructure/tagRepos:
```diff
@@ -0,0 +1,134 @@
+        echo "REPO_NAME: ${REPO_NAME}"
+
+        if (STATUS == 128) {
+            echo "Good, tag does not exist yet"
```
Where is it documented that `128` is returned in this situation? Do we need to be specific here?

AdamBrousseau replied:
I can't find it but I see 128 is used for other failure types too. So maybe we should just remove it and take any failure as "tag does not exist"

AdamBrousseau commented:
updated

AdamBrousseau commented:
Simplified script. Removed the two checks for existing tag and tag_point since having additional checks does not add any value. The git tag command will fail and stderr will indicate what is wrong.
